src/cli/Secrets.ts

```diff
@@ -1,34 +1,47 @@
+import process from 'process';
 import commander from 'commander';
-import { actionRunner, pkLogger, PKMessageType, determineNodePath } from '.';
+import { spawn, ChildProcess, exec } from 'child_process';
 import { PolykeyAgent } from '../lib/Polykey';
+import { actionRunner, pkLogger, PKMessageType, determineNodePath } from '.';
+
+
+const pathRegex = /^([a-zA-Z0-9_-]+)(?::)([a-zA-Z0-9_:-]+)(?:=)?([a-zA-Z_][a-zA-Z0-9_]+)?$/
 
 function makeListSecretsCommand() {
   return new commander.Command('list')
     .description('list all available secrets for a given vault')
     .alias('ls')
-    .requiredOption('-n, --vault-name <vaultName>', 'the vault name')
     .option('--node-path <nodePath>', 'node path')
     .option('--verbose', 'increase verbosity level by one')
+    .arguments('vault name(s) to list')
     .action(
       actionRunner(async (options) => {
         const client = PolykeyAgent.connectToAgent();
 
         const nodePath = determineNodePath(options);
         const isVerbose: boolean = options.verbose ?? false;
-        const vaultName: string = options.vaultName;
-        // Get list of secrets from pk
-        const secretNames = await client.listSecrets(nodePath, vaultName);
-
-        // List secrets
-        if (secretNames.length == 0) {
-          pkLogger(`no secrets found for vault '${vaultName}'`, PKMessageType.INFO);
-        } else {
-          if (isVerbose) {
-            pkLogger(`secrets contained within the ${vaultName} vault:`, PKMessageType.INFO);
+        const vaultNames: string[] = Array.from(options.args.values());
+
+        if (!vaultNames.length) {
+          throw Error('no vault names provided')
```

Comment:
We need to create a new exception heirarchy for the library and CLI commands. This will allow structured error handling for Polykey. Should be a new issue.

Reply:
this is also linked to https://github.com/MatrixAI/PolyKey/issues/51

Reply:
https://github.com/MatrixAI/js-polykey/issues/82